src/simulators/qasm/qasm_controller.hpp

```diff
@@ -383,6 +392,19 @@ OutputData QasmController::run_circuit(const Circuit &circ,
     case Method::statevector:
       if (simulation_precision_ == Precision::double_precision) {
         // Double-precision Statevector simulation
+#ifdef QASM_THRUST
```

Comment:
If we add GPU as a separate method then it could be a separate case here. Something like:

```cpp
case Method::statevector_gpu:
#ifdef QASM_THRUST
  // usual execution code
#else
  // throw an exception that GPU not enabled
#endif
```

The rest of the controller would then need to be updated so that checks for `Method::statevector` also check for `Method::statevector_gpu`

Reply:
implemented